rotkehlchen/tests/unit/test_inquirer.py

```diff
@@ -502,3 +503,29 @@ def mock_coingecko_return(url, *args, **kwargs):  # pylint: disable=unused-argum
         inquirer.find_usd_price(A_ETH, ignore_cache=True)
         # Check that the last price query contacted coingecko api
         assert coingecko_api_calls == 2
+
+
+@pytest.mark.parametrize('should_mock_current_price_queries', [False])
+@pytest.mark.parametrize('oracle', [CurrentPriceOracle.COINGECKO, CurrentPriceOracle.DEFILLAMA, CurrentPriceOracle.CRYPTOCOMPARE])  # noqa: E501
+def test_punishing_of_oracles_works(inquirer, oracle):
+    oracle_api_calls = 0
+
+    def mock_oracle_side_effect(url, *args, **kwargs):  # pylint: disable=unused-argument
+        nonlocal oracle_api_calls
+        oracle_api_calls += 1
+        raise requests.exceptions.RequestException('An unexpected error occured!')
+
+    oracle_instance = getattr(inquirer, f'_{oracle.name.lower()}')
+    oracle_patch = patch.object(oracle_instance.session, 'get', side_effect=mock_oracle_side_effect)  # noqa: E501
+    inquirer.set_oracles_order(oracles=[oracle])
+    with oracle_patch:
+        for _ in range(6):
+            assert inquirer.find_usd_price(A_ETH, ignore_cache=True) == Price(ZERO)
+    assert oracle_api_calls == 6
+
+    assert oracle_instance.is_penalized() is True
+    assert inquirer.find_usd_price(A_BTC, ignore_cache=True) == Price(ZERO)
+
+    with freeze_time(datetime.fromtimestamp(ts_now() + 5000)):
```

Comment:
Make this depend on the value of the variable

```suggestion
    with freeze_time(datetime.fromtimestamp(ts_now() + ORACLE_PENALTY_TS+1)):
```